pyproject.toml

```diff
@@ -19,6 +19,7 @@ dependencies = [
     'detoxify >= 0.5',
     'fugashi', # For tokenizer of metrics.ja.toxicity()
     'janome >= 0.3.1',
+    'hanlp >= 2.1.0b52',
```

Comment:
ERROR: Ignored the following yanked versions: 2.1.0a39, 2.1.0a40
ERROR: Could not find a version that satisfies the requirement hanlp>=2.1.0 (from langcheck[dev]) (from versions: 2.0.0a3, 2.0.0a4, 2.0.0a5, 2.0.0a6, 2.0.0a8, 2.0.0a9, 2.0.0a10, 2.0.0a11, 2.0.0a12, 2.0.0a13, 2.0.0a14, 2.0.0a15, 2.0.0a16, 2.0.0a17, 2.0.0a18, 2.0.0a19, 2.0.0a20, 2.0.0a21, 2.0.0a22, 2.0.0a24, 2.0.0a25, 2.0.0a26, 2.0.0a27, 2.0.0a28, 2.0.0a29, 2.0.0a30, 2.0.0a31, 2.0.0a32, 2.0.0a33, 2.0.0a34, 2.0.0a35, 2.0.0a36, 2.0.0a37, 2.0.0a38, 2.0.0a39, 2.0.0a40, 2.0.0a41, 2.0.0a42, 2.0.0a43, 2.0.0a44, 2.0.0a45, 2.0.0a46, 2.0.0a47, 2.0.0a48, 2.0.0a49, 2.0.0a50, 2.0.0a51, 2.0.0a52, 2.0.0a53, 2.0.0a54, 2.0.0a55, 2.0.0a56, 2.0.0a57, 2.0.0a58, 2.0.0a59, 2.0.0a60, 2.0.0a61, 2.0.0a62, 2.0.0a63, 2.0.0a64, 2.0.0a65, 2.0.0a66, 2.0.0a67, 2.0.0a68, 2.0.0a69, 2.1.0a0, 2.1.0a1, 2.1.0a2, 2.1.0a3, 2.1.0a4, 2.1.0a5, 2.1.0a6, 2.1.0a7, 2.1.0a8, 2.1.0a9, 2.1.0a10, 2.1.0a11, 2.1.0a12, 2.1.0a13, 2.1.0a14, 2.1.0a15, 2.1.0a16, 2.1.0a17, 2.1.0a18, 2.1.0a19, 2.1.0a20, 2.1.0a21, 2.1.0a22, 2.1.0a23, 2.1.0a24, 2.1.0a25, 2.1.0a26, 2.1.0a27, 2.1.0a28, 2.1.0a29, 2.1.0a30, 2.1.0a31, 2.1.0a32, 2.1.0a33, 2.1.0a34, 2.1.0a35, 2.1.0a36, 2.1.0a37, 2.1.0a38, 2.1.0a41, 2.1.0a42, 2.1.0a43, 2.1.0a44, 2.1.0a45, 2.1.0a46, 2.1.0a47, 2.1.0a48, 2.1.0a50, 2.1.0a51, 2.1.0a52, 2.1.0a53, 2.1.0a54, 2.1.0a55, 2.1.0a56, 2.1.0a57, 2.1.0a58, 2.1.0a59, 2.1.0a60, 2.1.0a61, 2.1.0a62, 2.1.0a63, 2.1.0a64, 2.1.0a65, 2.1.0b0, 2.1.0b1, 2.1.0b2, 2.1.0b3, 2.1.0b4, 2.1.0b5, 2.1.0b6, 2.1.0b7, 2.1.0b8, 2.1.0b9, 2.1.0b10, 2.1.0b11, 2.1.0b12, 2.1.0b13, 2.1.0b14, 2.1.0b15, 2.1.0b16, 2.1.0b17, 2.1.0b18, 2.1.0b19, 2.1.0b20, 2.1.0b21, 2.1.0b22, 2.1.0b23, 2.1.0b24, 2.1.0b25, 2.1.0b26, 2.1.0b27, 2.1.0b28, 2.1.0b29, 2.1.0b30, 2.1.0b31, 2.1.0b32, 2.1.0b33, 2.1.0b34, 2.1.0b35, 2.1.0b36, 2.1.0b37, 2.1.0b38, 2.1.0b39, 2.1.0b40, 2.1.0b41, 2.1.0b42, 2.1.0b43, 2.1.0b44, 2.1.0b45, 2.1.0b46, 2.1.0b47, 2.1.0b48, 2.1.0b49, 2.1.0b50, 2.1.0b51, 2.1.0b52)
ERROR: No matching distribution found for hanlp>=2.1.0

It would raise this error

Reply:
Ah I see, it looks like the 2.1 release is still a pre release (https://github.com/hankcs/HanLP/releases), so pip won't install a pre release by default. Do we need the 2.1 pre release version for the code to work, or will it still work on older release (e.g. 1.8)?

Reply:
I could harly say it can be installed.
1. HanLp 1.X itself is a java lib i think (hanlp 1.x)[https://github.com/hankcs/HanLP/tree/1.x/src]
2.  (pyhanlp)[https://pypi.org/project/pyhanlp/] the python API for hanlp 1.8 requires python<=3.8 but langcheck requires py>=3.8
3.  hanlp 1.x based on bigram language model + double array trie + dictionary. hanlp 2.X use sentenceTransformer. if langcheck cares about speed on non-GPU when process large amount of text,  hanlp 1.X would takes priority over hanlp 2.0.
3.  I haven't checked hanlp 1.X precision on tokenization task. From their formula's disscussion, a lot of people still using them in enterprise's projects, I think the precise itself would not be the problem, although not SOTA, because the core corpus hanlp 1.X used was 10+ years ago. I will check this, if the pyhanlp can be installed in py>=3.8

Reply:
I see, thanks for doing the investigation! Given that the current code doesn't work with pyhanlp, let's go with hanlp 2.1 beta release for now.

I did a bunch of testing and it seems like it starts working from `2.1.0.b33` (i.e. anything after this commit  https://github.com/hankcs/HanLP/commit/7616dbb4cf47916d293274d0511be4595221c26f).

```suggestion
    'hanlp >= 2.1.0b33',
```